R/check-mac.R

```diff
@@ -0,0 +1,71 @@
+#' Check macOS package
+#'
+#' This function works by bundling source package, and then uploading to
+#' <https://mac.r-project.org/macbuilder/submit.html>.  Once building is
+#' complete you'll receive a link to the built package in the email address
+#' listed in the maintainer field.
+#'
+#' @template devtools
+#' @inheritParams pkgbuild::build
+#' @param email An alternative email to use, default `NULL` uses the package
+#'   Maintainer's email.
+#' @param quiet If `TRUE`, suppresses output.
+#' @param ... Additional arguments passed to [pkgbuild::build()].
+#' @family build functions
+#' @return The url with the check results (invisibly)
+#' @export
+check_mac_release <- function(pkg = ".", args = NULL, manual = TRUE, email = NULL, quiet = FALSE, ...) {
```

Comment:
Currently it seems the only thing the public API allows is to check on R-release and the M1 macs. Though the response from the POST request indicates future plans to support more versions of R and architectures, e.g.

```
> httr::content(res)
$submitted
[1] TRUE

$id
[1] "1637073137-c5c2e7a4dcbedebf"

$package
[1] "glue"

$version
[1] "1.5.0.9000"

$rflavor
[1] "r-release"

$rfw
[1] "4.1-arm64"

$url
[1] "https://mac.R-project.org/macbuilder/results/1637073137-c5c2e7a4dcbedeb
f/"
```